### PR TITLE
Define RGB colors

### PR DESCRIPTION
**Pull request by Erovia**
Define RGB colors and pass them to the rgblight functions, instead of
defining multiple macros.

## Description
Instead of having multiple macros for each color, the PR defines the color twice (RGB/HSV) that can be passed to the appropriate functions.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Erovia** [comment]
In this commit, I've left the old color macros in order to not break anything.

In the end I'd like to get rid of them, so in an other commit I could fix up the affected keymaps. (only 15)

**Comment by Erovia** [comment]
@drashna what's your opinion on fixing up the keymaps and removing the macros altogether?

**Comment by mechmerlin** [comment]
> @drashna what's your opinion on fixing up the keymaps and removing the macros altogether?

I'd be in favor of doing that, but in a different PR.

**Comment by Erovia** [comment]
@mechmerlin thanks for the input. Let's do that. Once this is merged, I'll open another PR with the keymap fixes.

**Comment by drashna** [comment]
I'm not sure how I feel about it, actually.  But I'll post it in that PR.

**Comment by Erovia** [comment]
Agreed, although I've no way of testing it.  
Also fine-tuned the wording in the Docs and added the info to RGB Matrix's page.

File: quantum/rgb_matrix.c [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "eeprom.h"
 #include <string.h>
 #include <math.h>
+#include "rgblight_list.h"
```
Comment (drashna) [inline quantum/rgb_matrix.c]: Actually, this should go in `rgb_matrix.h`, that way, it's picked up everywhere.

Comment (Erovia) [reply quantum/rgb_matrix.c]: Thanks for the info.
Moved it to the header file.

**Comment by drashna** [comment]
Just remembering, I may need to walk my comment back about the rgb matrix include.  That or it needs to be updated/changed.

The RGB Light feature uses 0-360 (degrees) for hue, while the RGB Matrix uses 0-255 for hue.  This means that the values are NOT the same, but should be very close.  

However, I don't like "very close" here. I prefer "exact".  

This can be handled normally, but .... if you enable rgb light and rgb matrix, it creates a case that cannot be consolidated.

File: quantum/rgblight_list.h [hunk 2]
```diff
@@ -13,111 +13,163 @@
...
+#define RGB_TEAL         0x00, 0x80, 0x80
+#define RGB_CYAN         0x00, 0xFF, 0xFF
+#define RGB_AZURE        0x99, 0xf5, 0xFF
+#define RGB_BLUE         0x00, 0x00, 0xFF
+#define RGB_PURPLE       0x7A, 0x00, 0xFF
+#define RGB_MAGENTA      0xFF, 0x00, 0xFF
+#define RGB_PINK         0xFF, 0x80, 0xBF
+
+/*            HSV COLORS            */
+#define HSV_WHITE          0,   0, 255
```
Comment (drashna) [inline quantum/rgblight_list.h]: Could you add a `HSV_MATRIX_*` section here, as well? 

Specifically, the rgblight feature uses Hue properly (0-360) while rgb matrix uses incorrectly (0-255).  So using this for rgb matrix may be close, but it's not exact.

**Pull request merged**